app/models/topic.rb

```diff
@@ -830,6 +832,23 @@ def invite_by_email(invited_by, email, group_ids = nil, custom_message = nil)
     Invite.invite_by_email(email, invited_by, self, group_ids, custom_message)
   end
 
+  def invite_existing_muted?(target_user, invited_by)
+    # check if inviter is muted
+    return true if invited_by.id && MutedUser.where(user_id: target_user.id, muted_user_id: invited_by.id)
```

Comment:
Small nitpick here regarding the style. Usually we use the style `return true if something` only when all the code fits in a single line.